### ISSUE TITLE
Fix S1144 FP: Private Attributes

### ISSUE THREAD

**Issue by jeremia** [SonarSource/sonar-dotnet#8342]
### Description

Regarding S1144: Remove the unused class 'XXX'
For a private attribute class without AttributeUsageAttribute SonarLint always reports the class as unused even if it's used.

### Repro steps

namespace ConsoleApp;

```charp
public class Test
{
	[Testing]
	public static void Main()
	{
	}

	private sealed class TestingAttribute : Attribute
	{
	}
}
```

### Expected behavior

Only report CA1018: Specify AttributeUsage on XXX, Not S1144.

### Actual behavior

Both S1144 and CA1018 is reported

### Known workarounds

Att AttributeUsage: This removes both CA1018 (regarding attribute usage) and S1144 (regarding unused class).

### Related information

* C#/VB.NET Plugins version: SonarLint for Visual Studio 7.3.0.77872
* Visual Studio version: 17.7.6
* MSBuild / dotnet version: .NET 7
* SonarScanner for .NET version (if used): Not used
* Operating System: Windows 10

**Comment by antonioaversa** [comment]
Hi @jeremia, thanks for reporting this issue.

I can [reproduce](https://github.com/SonarSource/sonar-dotnet/pull/8354) it myself, and I can confirm it is indeed an issue that seems to appear when attributes are used.
We have recently encountered [similar issues](https://github.com/SonarSource/sonar-dotnet/issues/8348) with attributes on private methods. 
Your issue has been added to the backlog and will be included in one of our upcoming hardening sprints.

Meanwhile, waiting for a fix, you have the following options:
- If you are using SonarLint with SonarQube or SonarCloud in connected mode: you can mark the issue as a false-positive in the UI. The rule can also be disabled in your Quality Profile if it’s particularly noisy.
- If you are using SonarLint in Standalone Mode: you can use `#pragma warning disable` and `#pragma warning restore` to locally disable the rule in the source, or the `.editorconfig` files to disable or change the severity of the rule, if it gets particularly noisy.

Hope it helps,
Antonio

**Comment by martin-strecker-sonarsource** [comment]
@zsolt-kolbay-sonarsource please also change the title of the issue.
